tests/scripts/generate_psa_tests.py

```diff
@@ -227,9 +227,24 @@ def test_cases_for_not_supported(self) -> Iterator[test_case.TestCase]:
 class StorageKey(psa_storage.Key):
     """Representation of a key for storage format testing."""
 
-    def __init__(self, *, description: str, **kwargs) -> None:
+    def __init__(
+            self,
+            description: str,
+            expected_usage: Optional[str] = None,
+            **kwargs
+    ) -> None:
+        """Prepare to generate a key.
+
+        * `description`: used for the the test case names
+        * `implicit_usage`: the usage flags generated as the expected usage
+                            flags in the test cases. When testing implicit
+                            usage flags, they can differ in the generated keys
+                            and the expected usage flags in the test cases.
+        """
         super().__init__(**kwargs)
         self.description = description #type: str
+        self.usage = psa_storage.as_expr(expected_usage) if expected_usage is not None else\
+                     self.original_usage  #type: psa_storage.Expr
```

Comment:
If a key object have a single usage field then a new container class is needed to store the key and the expected usage flags because the expected usage flags in test cases can differ from the usage flags in the keys.

Reply:
A new class, yes. But it can and should be a derived class.